api/src/test/java/org/pmiops/workbench/dataset/DataSetMapperTest.java

```diff
@@ -0,0 +1,194 @@
+    dbDataDictionaryEntry.setFieldName("field");
+    dbDataDictionaryEntry.setOmopCdmStandardOrCustomField("field");
+    dbDataDictionaryEntry.setDescription("desc");
+    dbDataDictionaryEntry.setFieldType("type");
+    dbDataDictionaryEntry.setSourcePpiModule("s");
+    dbDataDictionaryEntry.setTransformedByRegisteredTierPrivacyMethods(false);
+  }
+
+  @Test
+  public void dbModelToClientLight() {
+    final DataSet toClientDataSet = dataSetMapper.dbModelToClientLight(dbDataset);
+    assertDbModelToClientLight(toClientDataSet, dbDataset);
+  }
+
+  @Test
+  public void dbModelToClientDataSet() {
+    final DataSet toClientDataSet = dataSetMapper.dbModelToClient(dbDataset);
+    assertDbModelToClient(toClientDataSet, dbDataset);
+  }
+
+  @Test
+  public void dbModelToClientDataDictionaryEntry() {
+    final DataDictionaryEntry toClientDataDictionaryEntry =
+        dataSetMapper.dbModelToClient(dbDataDictionaryEntry);
+    assertDbModelToClient(toClientDataDictionaryEntry, dbDataDictionaryEntry);
+  }
+
+  private void assertDbModelToClient(DataSet dataSet, DbDataset dbDataset) {
+    assertThat(dbDataset.getCohortIds())
+        .isEqualTo(dataSet.getCohorts().stream().map(Cohort::getId).collect(Collectors.toList()));
+    assertThat(dbDataset.getConceptSetIds())
+        .isEqualTo(
+            dataSet.getConceptSets().stream().map(ConceptSet::getId).collect(Collectors.toList()));
+    assertThat(dbDataset.getValues())
+        .isEqualTo(
+            dataSet.getDomainValuePairs().stream()
+                .map(
+                    dvp ->
+                        new DbDatasetValue(
+                            DbStorageEnums.domainToStorage(dvp.getDomain()).toString(),
+                            dvp.getValue()))
+                .collect(Collectors.toList()));
+    assertDbModelToClientLight(dataSet, dbDataset);
+  }
+
+  private void assertDbModelToClientLight(DataSet dataSet, DbDataset dbDataset) {
+    assertThat(dbDataset.getDataSetId()).isEqualTo(dataSet.getId());
+    assertThat(dbDataset.getVersion()).isEqualTo(Etags.toVersion(dataSet.getEtag()));
+    assertThat(dbDataset.getName()).isEqualTo(dataSet.getName());
+    assertThat(dbDataset.getIncludesAllParticipants())
+        .isEqualTo(dataSet.getIncludesAllParticipants());
+    assertThat(dbDataset.getDescription()).isEqualTo(dataSet.getDescription());
+    assertThat(dbDataset.getWorkspaceId()).isEqualTo(dataSet.getWorkspaceId());
+    assertThat(dbDataset.getLastModifiedTime().toInstant().toEpochMilli())
+        .isEqualTo(dataSet.getLastModifiedTime());
+    assertThat(dbDataset.getPrePackagedConceptSet())
+        .isEqualTo(
+            DbStorageEnums.prePackagedConceptSetsToStorage(dataSet.getPrePackagedConceptSet()));
+  }
+
+  private void assertDbModelToClient(
+      DataDictionaryEntry dataDictionaryEntry, DbDataDictionaryEntry dbDataDictionaryEntry) {
+    assertThat(dbDataDictionaryEntry.getCdrVersion().getCdrVersionId())
```

Comment:
Rather than individual elements since DataDictionaryEntry has equal and hashcode can we just have assertThat(dbDataDictionaryEntry).isEqual(dataDictionaryEntry)?

Reply:
dbDataDictionaryEntry and dataDictionaryEntry are different types. The equal and hashcode defined in DataDictionaryEntry is only for comparison to objects of type DataDictionaryEntry not DbDataDictionaryEntry

Reply:
Oh man i just missed that :D i thought its the same type DataDictionaryEntry